src/Microsoft.ML.AutoML/Sweepers/SweeperProbabilityUtils.cs

```diff
@@ -115,6 +118,10 @@ public static float[] ParameterSetAsFloatArray(IValueGenerator[] sweepParams, Pa
             return result.ToArray();
         }
 
+        private static T? GetIfIParameterValueOfT<T>(IParameterValue parameterValue)
+            where T : struct =>
+            parameterValue is IParameterValue<T> pvt ? pvt.Value : default(T?);
+
         public static ParameterSet FloatArrayAsParameterSet(IValueGenerator[] sweepParams, float[] array, bool expandedCategoricals = true)
         {
             Runtime.Contracts.Assert(array.Length == sweepParams.Length);
```

Comment:
_(comment pinned to a line of code to allow for threaded conversation)_

As mentioned by @antoniovs1029 (https://github.com/dotnet/machinelearning/pull/5163#issuecomment-639186293) a unit test setting the culture to pl-PL would be beneficial. Since the AutoML code currently uses the SMAC sweeper, duplicating this test should work:
https://github.com/dotnet/machinelearning/blob/f5aa7da43cea662f7b53737a436206138e0fee33/test/Microsoft.ML.Sweeper.Tests/TestSweeper.cs#L639-L691

If the CI servers are fast enough to complete >70 iterations in perhaps 30s, this unit test with a longer timeout would give an end-to-end test:
https://github.com/dotnet/machinelearning/blob/68ea9690ef6582a0decc974055c1f6982227aaf5/test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs#L105-L121

Info on why 70+ iterations are needed -- https://github.com/dotnet/machinelearning/pull/5455#discussion_r511429267

Reply:
Side question about setting culture -- when setting `Thread.CurrentThread.CurrentCulture = new CultureInfo("pl-PL")`, will sub-threads inherit the set culture? 

If not, we can verify that the sweep parameters are being written/read in the same thread with the culture set.

Reply:
To answer my own question, setting `Thread.CurrentThread.CurrentCulture = new CultureInfo("pl-PL")` does cause the current code to crash.

Repro from my earlier https://github.com/dotnet/machinelearning/issues/5320#issuecomment-663838031: https://dotnetfiddle.net/nWpCkP

That repro uses `CultureInfo.DefaultThreadCurrentCulture = new CultureInfo("pl-PL")` which sets the culture for all threads; `Thread.CurrentThread.CurrentCulture` seems to work equally.

Reply:
We do have a [unit test](https://github.com/dotnet/machinelearning/blob/b34c3b6ea5b0d7434dde63012c4c123a85d29dcb/test/Microsoft.ML.CodeGenerator.Tests/TrainerGeneratorTests.cs#L24-L50) in the AutoML CodeGen which tests "de-DE", but not the AutoML main call. And two other unit tests for specific cultures for [LineParser](https://github.com/dotnet/machinelearning/blob/b34c3b6ea5b0d7434dde63012c4c123a85d29dcb/test/Microsoft.ML.Tests/Transformers/LineParserTests.cs#L51-L74) & [LightGBM](https://github.com/dotnet/machinelearning/blob/ae1b7099472a7aef52b5186501166f58c3cc78d0/test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs#L749-L762).

Reply:
Thanks for looking into this, @justinormont 

The `TestSmacSweeper` test you mentioned doesn't seem to be affected by the changes on this PR. It seems that test is actually testing the Microsoft.ML.Sweeper project, whereas this PR is changing the behavior of Microsoft.ML.AutoML.SmacSweeper class. I.e. it seems to me we have 2 different SMAC sweepers (the AutoML project doesn't depend on the Microsoft.ML.Sweeper project).

The `AutoFitRegressionTest` failed after setting the culture to Russian _and_ the experiment time to 30s. But with the changes on this PR it passes the test correctly. I've opened PR #5462 to test this on the CI. I'd plan to merge this PR here and then merge that other PR to have the test.

Reply:
>  ..it seems to me we have 2 different SMAC sweepers (the AutoML project doesn't depend on the Microsoft.ML.Sweeper project).

Quite interesting. We modified the SMAC code when we pulled the AutoML code out to its own repo. But took on technical debt by _duplicating_ the SMAC code when we remerged the AutoML code to the main repo.

> I've opened PR #5462 to test this on the CI. I'd plan to merge this PR here and then merge that other PR to have the test.

SGTM; thanks for the unit test.

Reply:
Feel free to merge tests here if you prefer.

Reply:
Thanks.
#Resolved